File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int best = 0;
        int minPrice = prices[0];
        for (int i=1; i<(int)prices.size(); i++){
            minPrice = min(minPrice,prices[i]);
            best = max(best,prices[i]-minPrice);
        }
        return best;
    }
};

File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.py
class Solution(object):
    def maxProfit(self, prices):
        minPrice  = None
        maxProfit = 0
        for price in prices:
            if minPrice is None: minPrice = price
            maxProfit = max(maxProfit, max(price - minPrice,0))
            minPrice  = min(minPrice, price)
        return maxProfit
        